src/internal/m365/service/onedrive/restore.go

```diff
@@ -110,6 +110,10 @@ func AugmentRestorePaths(
 	colPaths := map[string]path.RestorePaths{}
 
 	for _, p := range paths {
+		if p.StoragePath.Category() == path.ChannelMessagesCategory {
```

Comment:
Looks good. We should probably enhance the test for `AugmentRestorePaths` though.

Reply:
Working on creating dataset for the testcase.

Reply:
This doesn't feel like the right ownership.  `/service/onedrive` should never see a ChannelMessage category path.  Are we sure this is in the right place?  Alternatively, are we sure there's not a more severe bug, of which this is only a symptom?

My assumption is that either: 1/ `AugmentRestorePaths` is owned by the wrong package and needs to be moved to a more generic location or 2/ this should be an inclusive check (ex: `if not category files/folders then continue`) instead of an exclusive one (ie: `allow everything except channel messages`).  2/ is probably the more correct solution.